### DISCUSSION TITLE
Where is the best place to load UI assets like button images?

### DISCUSSION THREAD

**Discussion by kuolas** [emilk/egui#4445] [Q&A]
I am creating a image manipulation program, like Gimp, with a toolbar with icons with text.

I load the button images with this code:
`let crop_icon = egui::Image::new(egui::include_image!("../assets/tool-icons/crop.svg")).tint(self.sys_color);`
            
          
And use it like this:
```
impl eframe::App for MyApp {

    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {

        let crop_icon = egui::Image::new(egui::include_image!("../assets/tool-icons/crop.svg"))
            .tint(self.sys_color);

            egui::menu::bar(ui, |ui| {
                if ui
                    .add(egui::Button::image_and_text(crop_icon, "Crop"))
                    .clicked()
                {}
            });
        });
        
}
```

But I do not now if loading my assets in the update function is correct.

- What is the correct way to load icons for the UI
- Should I create a function that loads assets into a struct and then use that struct in the app?

Any recommendations are welcome!

**Comment by metasim** [comment]
I don't think where you load it matters when it's static image (as opposed to a dynamic texture), as the data gets cached:

https://github.com/emilk/egui/blob/738ea75453567c5f17a543e68aec8c48097cae7b/crates/egui/src/load/bytes_loader.rs#L34
